src/back/spv/writer.rs

```diff
@@ -1225,6 +1225,23 @@ impl Writer {
                 };
 
                 self.decorate(id, Decoration::BuiltIn, &[built_in as u32]);
+
+                use crate::ScalarKind as Sk;
+                use crate::TypeInner as Ty;
+                // Spirv spec: Any variable with integer or double-precision floating-point
```

Comment:
Do cite the specific VUID here - that's meant to be a stable identifier for this requirement. Also, it's not part of the SPIR-V spec per se; rather, it's a requirement that Vulkan imposes on the SPIR-V it accepts. (The SPIR-V spec refers to this as the "client API". SPIR-V is also used as the target for OpenCL, which has to run arbitrary C++, so its client API has much looser requirements.)